modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -738,12 +741,35 @@ protected void parseFromTextExpression(String key, DslElementSyntax paramDsl, Su
     return definitionBuilder;
   }
 
+  protected void parseObjectParameter(String key, String name, MetadataType type, DslElementSyntax dsl,
+                                      Object defaultValue,
+                                      ExpressionSupport expressionSupport, boolean required,
+                                      Set<ModelProperty> modelProperties,
+                                      List<StereotypeModel> allowedStereotypes, boolean content) {
+    if (content) {
+      parseFromTextExpression(name, dsl, () -> value -> valueResolverFactory.of(name,
+                                                                                type,
+                                                                                value,
+                                                                                defaultValue,
+                                                                                expressionSupport,
+                                                                                required,
+                                                                                modelProperties,
+                                                                                !allowedStereotypes.isEmpty(),
+                                                                                content));
+    } else {
+      parseObjectParameter(key, name, (ObjectType) type,
```

Comment:
shouldn't this one be renamed to remain consistent with the change in line 772?

Reply:
actually, this one does roefer to an object type because this is in the `else` of an `if(content)`